lib/archethic/p2p.ex

```diff
@@ -152,11 +164,27 @@ defmodule Archethic.P2P do
   end
 
   @doc """
-  Determine if the node public key is authorized and available
+  Determine if the node public key is authorized and available for the given datetime (default to now) or before if needed
   """
-  @spec authorized_and_available_node?(Crypto.key()) :: boolean()
-  def authorized_and_available_node?(node_public_key \\ Crypto.first_node_public_key()) do
-    Utils.key_in_node_list?(authorized_and_available_nodes(), node_public_key)
+  @spec authorized_and_available_node?(
+          Crypto.key(),
+          datetime :: DateTime.t(),
+          before? :: boolean()
+        ) :: boolean()
+  def authorized_and_available_node?(
```

Comment:
Here there is an edge case you don't handle, it's when the first node bootstrap. At this state no nodes are really authorized nor available. So the first node cannot send any transaction until the first the next authorization date.
The previous function was returning the first enrolled node in case no nodes are available and authorized (see `authorized_and_available_nodes` function)